pkg/cmd/server/origin/master.go

```diff
@@ -390,10 +393,24 @@ func (c *MasterConfig) InitializeObjects() {
 	c.ensureOpenShiftSharedResourcesNamespace()
 }
 
-func (c *MasterConfig) InstallProtectedAPI(apiContainer *genericmux.APIContainer) ([]string, error) {
+func (c *MasterConfig) InstallProtectedAPI(apiserver *genericapiserver.GenericAPIServer) ([]string, error) {
+	apiContainer := apiserver.HandlerContainer
+
 	// initialize OpenShift API
 	storage := c.GetRestStorage()
 
+	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(buildapi.GroupName)
+	//if apiResourceConfigSource.AnyResourcesForVersionEnabled(rbacapiv1alpha1.SchemeGroupVersion) {
+	apiGroupInfo.VersionedResourcesStorageMap[buildapiv1.SchemeGroupVersion.Version] = map[string]rest.Storage{
+		"builds":         storage["builds"],
+		"builds/clone":   storage["builds/clone"],
+		"builds/log":     storage["builds/log"],
+		"builds/details": storage["builds/details"],
```

Comment:
need buildConfigs, etc. I wonder if we want to move this somewhere more visible or have a test that checks if we did this for all storage endpoints (in case you add new endpoint and forged to register it).